vicodyn/src/vicodyn/balancer/simple.cpp

```diff
@@ -14,16 +14,23 @@ simple_t::simple_t(context_t& ctx, peers_t& peers, asio::io_service& loop, const
     peers(peers),
     logger(ctx.log(format("balancer/simple/{}", app_name))),
     args(args),
-    _retry_count(args.as_object().at("retry_count", 4u).as_uint()),
     app_name(app_name),
     x_cocaine_cluster(locator_extra.at("x-cocaine-cluster", "").as_string()),
+    attempt_limit(args.as_object().at("attempts", dynamic_t::empty_object).as_object().at("simple", 2U) .as_uint()),
     ban_timeout(args.as_object().at("ban-timeout-ms", 0U).as_uint())
 {
     COCAINE_LOG_INFO(logger, "created simple balancer for app {}", app_name);
 }
 
 auto simple_t::choose_peer(const std::shared_ptr<request_context_t>& request_context,
                 const hpack::headers_t& /*headers*/, const std::string& /*event*/) -> std::shared_ptr<peer_t> {
+    static const std::string simple_counter_name = "simple_attempts";
+
+    auto& attempts = request_context->counter(simple_counter_name);
+    if (attempts >= attempt_limit) {
+        throw error_t("count of attempts is reached");
```

Comment:
Should be this exception reported to log? If yes, it is better to report additionally an`attempts_limit`.

Reply:
Ok